src/borg/repository.py

```diff
@@ -650,18 +650,25 @@ def report_error(msg):
         try:
             transaction_id = self.get_transaction_id()
             current_index = self.open_index(transaction_id)
-        except Exception:
+            logger.debug('Read committed index of transaction %d', transaction_id)
+        except Exception as exc:
             transaction_id = self.io.get_segments_transaction_id()
             current_index = None
+            logger.debug('Failed to read committed index (%s), using segments transaction %s', exc, transaction_id)
```

Comment:
maybe move this to "else" branch of the following "if"?
you could also use %d then for the not-None transaction_id.